knoten/bundle.py

```diff
@@ -342,6 +338,43 @@ def compute_parameter_weights(network, sensors, parameters, coefficient_columns)
 
     return weight_mat
 
+def compute_image_weight(sensor, parameters):
+    """
+    Compute the weight matrix for the sensor parameters of a single image.
+
+    Parameters
+    ----------
+    sensor : csmapi.RasterGm
+             The CSM sensor model for the image
+    parameters : list
+                 The list of parameters to solve for
+    """
+    param_count = len(parameters)
+    covar_mat = np.zeros((param_count, param_count))
+    for a, b in itertools.product(range(param_count), range(param_count)):
+        covar_mat[a, b] = sensor.getParameterCovariance(parameters[a].index, parameters[b].index)
+    return np.linalg.inv(covar_mat)
```

Comment:
Does this need  a 'default'/backup option like `compute_point_weight` which returns the identity matrix if needed?

Reply:
This doesn't have a default because part of the CSM spec is that sensor models should always be able to provide these. For our control networks, only constrained points have co-variance matrices, so it is realistic for a point to not have a covariance matrix.